tests/subsys/power/power_mgmt_soc/src/power_mgmt.c

```diff
@@ -0,0 +1,337 @@
+		return;
+	}
+
+	if (sys_pm_is_sleep_state(state))
+		pm_counters[0].exit_cnt++;
+	else if (sys_pm_is_deep_sleep_state(state))
+		pm_counters[1].exit_cnt++;
+}
+
+static void pm_check_counters(uint8_t cycles)
+{
+	for (int i = 0; i < SLP_STATES_SUPPORTED; i++) {
+
+		LOG_INF("PM state[%d] entry counter %d\n", i,
+			pm_counters[i].entry_cnt);
+		LOG_INF("PM state[%d] exit counter %d\n", i,
+			pm_counters[i].exit_cnt);
+
+		zassert_true(pm_counters[i].entry_cnt
+				== pm_counters[i].exit_cnt,
+				"PM counters entry/exit mismatch");
+
+		pm_counters[i].entry_cnt = 0;
+		pm_counters[i].exit_cnt = 0;
+	}
+}
+
+static void pm_reset_counters(void)
+{
+	for (int i = 0; i < SLP_STATES_SUPPORTED; i++) {
+		pm_counters[i].entry_cnt = 0;
+		pm_counters[i].exit_cnt = 0;
+	}
+
+	checks_enabled = false;
+}
+
+static void pm_trigger_marker(void)
+{
+	trigger_time = k_uptime_get();
+
+	printk("PM >\n");
+}
+
+static void pm_exit_marker(void)
+{
+	int64_t residency_delta;
+	int secs;
+	int msecs;
+
+	printk("PM <\n");
+
+	if (trigger_time > 0) {
+		residency_delta = k_uptime_delta(&trigger_time);
+		secs = (int)(residency_delta / SEC_TO_MSEC);
+		msecs = (int)(residency_delta % SEC_TO_MSEC);
+		LOG_INF("PM sleep residency %d.%03d seconds", secs, msecs);
+	}
+}
+
+static int task_a_init(void)
+{
+	LOG_INF("Thread task A init");
+
+	return 0;
+}
+
+static int task_b_init(void)
+{
+	LOG_INF("Thread task B init");
+
+	return 0;
+}
+
+void task_a_thread(void *p1, void *p2, void *p3)
+{
+	while (true) {
+		k_msleep(THREAD_A_SLEEP_TIME);
+		printk("A");
+	}
+}
+
+static void task_b_thread(void *p1, void *p2, void *p3)
+{
+	while (true) {
+		k_msleep(THREAD_B_SLEEP_TIME);
+		printk("B");
+	}
+}
+
+static void create_tasks(void)
+{
+	task_a_init();
+	task_b_init();
+
+	k_thread_create(&thread_a_id, stack_a, TASK_STACK_SIZE, task_a_thread,
+		NULL, NULL, NULL, PRIORITY,  K_INHERIT_PERMS, K_FOREVER);
+	k_thread_create(&thread_b_id, stack_b, TASK_STACK_SIZE, task_b_thread,
+		NULL, NULL, NULL, PRIORITY,  K_INHERIT_PERMS, K_FOREVER);
+
+	k_thread_start(&thread_a_id);
+	k_thread_start(&thread_b_id);
+
+}
+
+static void destroy_tasks(void)
+{
+	k_thread_abort(&thread_a_id);
+	k_thread_abort(&thread_b_id);
+
+	k_thread_join(&thread_a_id, K_FOREVER);
+	k_thread_join(&thread_b_id, K_FOREVER);
+}
+
+static void suspend_all_tasks(void)
+{
+	k_thread_suspend(&thread_a_id);
+	k_thread_suspend(&thread_b_id);
+}
+
+static void resume_all_tasks(void)
+{
+	k_thread_resume(&thread_a_id);
+	k_thread_resume(&thread_b_id);
+}
+
+int test_pwr_mgmt_multithread(uint8_t cycles)
+{
+	uint8_t iterations = cycles;
+
+	create_tasks();
+
+	LOG_INF("PM multi-thread test started for cycles: %d", cycles);
+
+	checks_enabled = true;
+	while (iterations-- > 0) {
+
+		/* Light sleep cycle */
+		LOG_INF("Suspend...");
+		suspend_all_tasks();
+		LOG_INF("About to enter light sleep");
+		pm_trigger_marker();
+		k_msleep(CONFIG_SYS_PM_MIN_RESIDENCY_SLEEP_1 +
+			 LT_EXTRA_SLP_TIME);
+		k_busy_wait(100);
```

Comment:
Why are these 100 us busy waits being used after the sleeps?

Reply:
Just put a real quick busy spinning after sleeping to avoid errors. I could remove it if it's considered unnecessary.

Reply:
Also, I removed the userspace configuration. The LOG_PROCESS_THREAD_SLEEP_MS is set for micro chip, otherwise the latency would exceed expected values if using default. Ignore the warning for cc1352 or cc26x2.

Reply:
> Just put a real quick busy spinning after sleeping to avoid errors. I could remove it if it's considered unnecessary.

Why are there errors if it's not there?  That seems like something that needs to be understood.

Reply:
I mean I put it there just for safety. I removed it and ran it on micro chip and there were no errors.

Reply:
That part was good.  Now just have to deal with this build warning from:

     rm -rf build && west build -b cc1352r1_launchxl


```
warning: LOG_PROCESS_THREAD_SLEEP_MS (defined at subsys/logging/Kconfig:200) was assigned the value
'2100' but got the value ''. Check these unsatisfied dependencies: LOG_PROCESS_THREAD (=n),
(!LOG_MINIMAL) (=n). See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_LOG_PROCESS_THREAD_SLEEP_MS.html
and/or look up LOG_PROCESS_THREAD_SLEEP_MS in the menuconfig/guiconfig interface. The Application
Development Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of
the manual might be helpful too.
```

I.e. remove that config option from `proj.conf`, or perhaps disable logging since that might affect the sleep behavior and should not affect the test.

Reply:
Removed in the latest commit.